File: core/src/net/natruid/jungle/views/AbstractView.java
package net.natruid.jungle.views;

import com.badlogic.gdx.scenes.scene2d.Stage;
import net.natruid.jungle.core.Sky;

public abstract class AbstractView extends Stage {
    public AbstractView() {
        super(Sky.jungle.getUiViewport());
    }

    public void render(float delta) {
        this.act(delta);
        this.draw();
    }

    public void show() {
    }

    public void hide() {
    }

    public void pause() {
    }

    public void resume() {
    }
}

File: core/src/net/natruid/jungle/views/AbstractView.kt
package net.natruid.jungle.views

import com.badlogic.gdx.scenes.scene2d.Stage
import net.natruid.jungle.core.Jungle

abstract class AbstractView : Stage(Jungle.instance.uiViewport) {
    open fun render(delta: Float) {
        act(delta)
        draw()
    }

    open fun show() {}

    open fun hide() {}

    open fun pause() {}

    open fun resume() {}
}
